File: flutter_app/lib/models/booking_slot.dart
class BookingSlotModel {
  final int id;
  final int storeId;
  final String slotDate;
  final String slotTime;
  final int maxBookings;
  final int currentBookings;
  final bool isAvailable;

  BookingSlotModel({
    required this.id,
    required this.storeId,
    required this.slotDate,
    required this.slotTime,
    required this.maxBookings,
    required this.currentBookings,
    required this.isAvailable,
  });

  bool get isFull => currentBookings >= maxBookings;

  factory BookingSlotModel.fromJson(Map<String, dynamic> json) => BookingSlotModel(
        id: json['id'],
        storeId: json['store_id'],
        slotDate: json['slot_date'],
        slotTime: json['slot_time'],
        maxBookings: json['max_bookings'],
        currentBookings: json['current_bookings'],
        isAvailable: json['is_available'] == true,
      );
}

File: app/Models/BookingSlot.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class BookingSlot extends Model
{
    protected $fillable = [
        'store_id', 'slot_date', 'slot_time',
        'max_bookings', 'current_bookings', 'is_available',
    ];

    protected $casts = [
        'slot_date'    => 'date',
        'is_available' => 'boolean',
    ];

    public function store()
    {
        return $this->belongsTo(Store::class);
    }

    public function bookings()
    {
        return $this->hasMany(Booking::class);
    }

    public function isFull(): bool
    {
        return $this->current_bookings >= $this->max_bookings;
    }
}
